torch/onnx/verify.py

```diff
@@ -0,0 +1,262 @@
+import copy
+import io
+import os
+import tempfile
+from contextlib import ExitStack
+from typing import Optional, Union
+
+import numpy as np
+
+import torch
+from torch import Tensor
+from torch.onnx.utils import unpack_quantized_tensor
+
+_ORT_PROVIDERS = ["CPUExecutionProvider"]
+
+
+def _flatten_tuples(elem):
+    flattened = []
+    for t in elem:
+        if isinstance(t, tuple):
+            flattened.extend(_flatten_tuples(t))
+        else:
+            flattened.append(t)
+    return flattened
+
+
+def _to_numpy(elem):
+    if isinstance(elem, Tensor):
+        if elem.requires_grad:
+            return elem.detach().cpu().numpy()
+        else:
+            return elem.cpu().numpy()
+    elif isinstance(elem, (list, tuple)):
+        return [_to_numpy(inp) for inp in elem]
+    elif isinstance(elem, bool):
+        return np.array(elem, dtype=bool)
+    elif isinstance(elem, int):
+        return np.array(elem, dtype=int)
+    elif isinstance(elem, float):
+        return np.array(elem, dtype=float)
+    elif isinstance(elem, dict):
+        flattened = []
+        for k in elem:
+            flattened += [_to_numpy(k)] + [_to_numpy(elem[k])]
+        return flattened
+    return elem
+
+
+def _convert_to_onnx(
```

Comment:
I recommend type annotating this function. It is unclear what some of the params defaulted to None should take.